pkg/regopolicyinterpreter/regopolicyinterpreter.go

```diff
@@ -0,0 +1,683 @@
+package regopolicyinterpreter
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sync"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/storage/inmem"
+	"github.com/open-policy-agent/opa/topdown"
+	"github.com/pkg/errors"
+)
+
+type LogLevel int
+
+const (
+	LogNone LogLevel = iota
+	// Logs the output of Rego print() statements in the policy
+	LogInfo
+	// Logs the result objects returned from each query
+	LogResults
+	// Logs the full metadata state after each query
+	LogMetadata
+)
+
+type RegoPolicyInterpreter struct {
+	// Mutex to prevent concurrent access to fields
+	mutex sync.Mutex
+	// Rego which describes policy behavior (see above)
+	code string
+	// Rego data namespace
+	data map[string]interface{}
+	// Modules
+	modules map[string]*RegoModule
+	// Compiled modules
+	compiledModules *ast.Compiler
+	// Logging
+	logLevel       LogLevel
+	logFile        *os.File
+	infoLogger     *log.Logger
+	resultsLogger  *log.Logger
+	metadataLogger *log.Logger
+}
+
+type RegoModule struct {
+	// The Rego namespace of the module
+	Namespace string
+	// The feed from which the module was obtained
+	Feed string
+	// The issuer of the module
+	Issuer string
+	// The module Rego code
+	Code string
+}
+
+/**
+Each rule can optionally return a series of metadata commands in addition to
```

Comment:
I feel like this amount and type of data should be in a README instead of as a block comment.

Reply:
Great point. I've moved it to a README as suggested.